File: ProtocolTestManager/PTMService/PTMService/ClientApp/src/model/RuleGroup.ts
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.
export const AllNode = {
  value: 'All',
  label: '(Select all)'
}

export interface Rule {
    DisplayName?: string;
    Name: string;
    Rules?: Rule[];
    Categories?: string[]
    MappingRules?: string[]
}

export interface RuleGroup {
    DisplayName?: string;
    Name: string;
    Rules: Rule[]
}

export interface RuleData {
    AllRules: RuleGroup[];
    SelectedRules: Rule[];
    TargetFilterIndex: number;
    MappingFilterIndex: number;
}

export interface SelectedRuleGroup {
    Name: string;
    Selected: string[]
}

export interface GroupItem {
    key: string;
    name: string;
    rules: Rule[];
    checked: string[];
    action: (data: any) => void;
}

File: ProtocolTestManager/PTMService/Common/Types/RuleGroup.cs
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using System.Collections.Generic;

namespace Microsoft.Protocols.TestManager.PTMService.Common.Types
{
    public class RuleGroup
    {
        public string DisplayName { get; set; }

        public string Name { get; set; }

        public IList<Rule> Rules { get; set; }
    }
}
